lib/resty/iprepd.lua

```diff
@@ -34,33 +34,73 @@ function _M.new(options)
     statsd_client = statsd
   end
 
+  local whitelist = nil
+  local whitelist_list = options.whitelist or nil
+  if whitelist_list then
+    whitelist = iputils.parse_cidrs(whitelist_list)
+  end
+
   local self = {
     url = iprepd_url,
+    timeout = options.timeout or 10,
     threshold = iprepd_threshold,
     api_key_hdr = {
       ['Authorization'] = 'APIKey ' .. iprepd_api_key,
     },
-    cache_ttl = cache_ttl,
-    timeout = options.timeout or 10,
     cache = cache,
+    cache_ttl = options.cache_ttl or 30,
     cache_errors = options.cache_errors or 0,
     statsd = statsd_client,
     statsd_host = options.statsd_host,
     statsd_port = options.statsd_port or 8125,
     statsd_max_buffer_count =  options.statsd_max_buffer_count or 100,
+    statsd_flush_timer = options.statsd_flush_timer or 5,
+    dont_block = options.dont_block or 0,
+    whitelist = whitelist,
   }
 
   return setmetatable(self, mt)
 end
 
 function _M.check(self, ip)
-  local httpc = http.new()
-  -- set timeout in ms
-  httpc:set_timeout(self.timeout)
+  if self.whitelist then
+    if iputils.ip_in_cidrs(ip, self.whitelist) then
+      return
+    end
+  end
+
+  local reputation = self:get_reputation(ip)
+  if reputation then
+    ngx.req.set_header('X-Foxsec-IP-Reputation', tostring(reputation))
+    if reputation <= self.threshold then
+      ngx.req.set_header('X-Foxsec-IP-Reputation-Below-Threshold', 'true')
+      ngx.req.set_header('X-Foxsec-Block', 'true')
+      if self.statsd then
+        self.statsd.incr("iprepd.status.rejected")
+      end
 
-  -- Get reputation for ip
+      if self.dont_block == 1 then
+        ngx.log(ngx.ERR, '[logonly] ' .. ip .. ' rejected with a reputation of ' .. reputation)
```

Comment:
Should we return after this line, so the subsequent header sets (for below-threshold to false after this code block) are not executed?

Reply:
At the moment, I'm thinking that we should always set the headers. I think I should change this log message to be a little more explicit. As well, should change the name of `self.statsd.incr("iprepd.status.rejected")`

Reply:
👍

Reply:
Oh haha, I need to drink more coffee before, you are completely right. I misunderstood the comment.